src/main/java/com/firebolt/jdbc/resultset/compress/LZ4InputStream.java

```diff
@@ -159,6 +160,12 @@ public byte[] readNextBlock() throws IOException {
 		// checksum - 16 bytes.
 		readFully(dataWrapper, checksum, 1, 15);
 		BlockChecksum expected = BlockChecksum.fromBytes(checksum);
+
+		// ALB keepalive
+		if (expected.equals(keepalive)) {
```

Comment:
Lets read a special magic byte that signifies keepalive. And a checksum that is `keepalivkeepaliv`

Reply:
sg, though `keepalivkeepaliv` is of 17 bytes :rofl: